drivers/source/usb/USBDevice.cpp

```diff
@@ -852,8 +853,17 @@ void USBDevice::ep0_setup()
     assert_locked();
 
     if (_device.state < Default) {
-        MBED_ASSERT(0);
+#if defined(MBED_TRAP_ERRORS_ENABLED) && MBED_TRAP_ERRORS_ENABLED
```

Comment:
Don't need to do this - just `#if MBED_TRAP_ERRORS_ENABLED` will do. If a macro isn't defined, it's treated as `0` in a `#if`.

(I have seen compilers that would warn about that, but then they'd still complain about using an undefined thing in a full expression like that).

Reply:
I just copied that from [a different file](https://github.com/ARMmbed/mbed-os/blob/7cbce82ed79888b0338ec0706e582be64e8f58f8/platform/source/mbed_retarget.cpp#L713) just in case there was something I was not aware of.

Reply:
It doesn't matter, it's harmless. The one pattern you might see but shouldn't follow though is

     #ifdef MBED_CONF_XXX_STATS_ENABLED

with those options defaulting to `null` in the JSON. Doing that meant setting the option to `false` would enable it. Boolean options should always be `true` or `false` in JSON, and tested with `#if`. `null` has its uses, but just being a odd spelling of `false` isn't one of them.

Reply:
This [force-push](https://github.com/ARMmbed/mbed-os/compare/eb5574dd06b4ffee31a36d511c581d3d5dec82fc..e58f40b7c7a9a5a44b2bb27fb6a4d735771f7af6) replaces the contional pre-processor directive with a simpler one.